alicloud/service_alicloud_ecs.go

```diff
@@ -806,3 +806,47 @@ func (s *EcsService) DescribeSnapshotById(snapshotId string) (*ecs.Snapshot, err
 	}
 	return &resp.Snapshots.Snapshot[0], nil
 }
+
+func (s *EcsService) DescribeSnapshotPolicy(id string) (*ecs.AutoSnapshotPolicy, error) {
+	args := ecs.CreateDescribeAutoSnapshotPolicyExRequest()
+	args.AutoSnapshotPolicyId = id
+
+	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DescribeAutoSnapshotPolicyEx(args)
+	})
+	if err != nil {
+		return nil, WrapErrorf(err, DefaultErrorMsg, args.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+
+	resp := raw.(*ecs.DescribeAutoSnapshotPolicyExResponse)
+	if len(resp.AutoSnapshotPolicies.AutoSnapshotPolicy) < 1 {
+		return nil, WrapErrorf(GetNotFoundErrorFromString(fmt.Sprintf("ECS DescribeSnapshotPolicy(id: %s)", id)), NotFoundMsg, args.GetActionName(), AlibabaCloudSdkGoERROR)
```

Comment:
Line 823 should be `return nil, WrapErrorf(Error(GetNotFoundMessage("SnapshotPolicy", id)), NotFoundMsg, args.GetActionName(), ProviderError)`